packages/typespec-python/test/generated/azure-core-basic/_specs_/azure/core/basic/_operations/_operations.py

```diff
@@ -346,7 +346,7 @@ def create_or_update(self, id: int, resource: Union[_models.User, JSON, IO], **k
         if isinstance(resource, (IOBase, bytes)):
             _content = resource
         else:
-            _content = json.dumps(resource, cls=AzureJSONEncoder)  # type: ignore
+            _content = json.dumps(resource, cls=AzureJSONEncoder, exclude_readonly=True, exclude_none=False)  # type: ignore
```

Comment:
I don't think we should replace it with `as_dict` bc:
1. some body is not `Model`
2. body could be a `MutableMapping[str, Any]` that mix simple `dict` and `Model`
3. `json.dumps(Model.as_dict())` will go through all models' attributes, it is not effiecient

Reply:
cc @iscai-msft @lmazuel @johanste